src/components/Main/Main.tsx

```diff
@@ -33,31 +34,36 @@ const Main: React.FC = () => {
   const searchTitle: string = useSelector<RootState, string>(
     searchedFilmSelector
   );
+  const offset: number = useSelector<RootState, number>(offsetSelector);
 
   const [selectedGenre, setGenre] = useState(DefaultFilters.defaultGenre);
 
   const handleChangeGenre = (event: React.MouseEvent) => {
-    const genre = (event.target as HTMLInputElement).id;
-
-    dispatch(changeGenre(genre));
-    setGenre(genre);
+    const genreChenged = (event.target as HTMLInputElement).id;
+    if (genre !== genreChenged) {
+      dispatch(clearfilmsList());
+      dispatch(changeGenre(genreChenged));
+      setGenre(genreChenged);
+    }
   };
 
   const handlechangeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    dispatch(changeSorting(event.target.value));
+    if (event.target.value !== sortingType) {
+      dispatch(clearfilmsList());
+      dispatch(changeSorting(event.target.value));
+    }
   };
 
   useEffect(() => {
     dispatch(
       fetchfilmsList.request({
         genre: genre,
         searchTitle: searchTitle,
+        offset: offset,
+        sortingType: sortingType,
```

Comment:
genre,
searchTitle,
...

Reply:
and make 1 saga task that is listening 3 different actions, filter, sort, search